### PR TITLE
Added Granular Control Over `PipeFile`s

### PR DESCRIPTION
**Pull request by danbugs**

### CODE REVIEW

File: containerd-shim-spin-v1/src/main.rs [hunk 1]
```diff
@@ -63,61 +63,53 @@ impl Wasi {
...
                         .read(true)
                         .write(true)
                         .open(stdin_pipe_path.clone())
                         .unwrap(),
                     stdin_pipe_path.clone(),
-                ),
-                PipeFile::new(
+                ))
+            },
+            if stdout_pipe_path.as_os_str().is_empty() {
```
Comment (Mossaka) [inline containerd-shim-spin-v1/src/main.rs]: maybe extract this to a utility function?

File: containerd-shim-spin-v1/src/main.rs [hunk 2]
```diff
@@ -63,61 +63,53 @@ impl Wasi {
...
-            || stdout_pipe_path.as_os_str().is_empty()
-            || stderr_pipe_path.as_os_str().is_empty()
-        {
-            None
-        } else {
-            Some(CustomLogPipes::new(
-                PipeFile::new(
+        let custom_log_pipes = CustomLogPipes::new(
+            if stdin_pipe_path.as_os_str().is_empty() {
+                None
```
Comment (Mossaka) [inline containerd-shim-spin-v1/src/main.rs]: like this!

File: Cargo.lock [hunk 3]
```diff
@@ -2285,7 +2285,7 @@ dependencies = [
 [[package]]
 name = "outbound-redis"
 version = "0.2.0"
-source = "git+https://github.com/danbugs/spin?branch=danbugs/custom-logging-pipes#d79eaf0cdcd4c71982ea8eb2f97e9c477b2acb3a"
+source = "git+https://github.com/danbugs/spin?branch=danbugs/custom-logging-pipes#635f4d73b25884d27351d1862aae6a58096810bb"
```
Comment (Mossaka) [inline Cargo.lock:2288]: can you add `rev` to the spin dependency in `Cargo.toml`

**Review by Mossaka — CHANGES_REQUESTED** [review]

File: Cargo.lock (hunk 3 above)
Comment (danbugs) [reply Cargo.lock:2288]: We are currently using my fork of spin. As I'm updating it kind of constantly while the PR is under review, I've avoided using a rev to save me the trouble of changing it all the time.

File: containerd-shim-spin-v1/src/main.rs (hunk 1 above)
Comment (danbugs) [reply containerd-shim-spin-v1/src/main.rs]: done ✅

**Review by Mossaka — APPROVED** [review]

**Pull request merged**